src/lib.rs

```diff
@@ -727,13 +677,14 @@ impl GooseAttack {
         let config = self.configuration.clone();
         loop {
             for task_sets_index in &weighted_task_sets {
+                let base_url = goose::get_base_url(
+                    Some(config.host.to_string()),
+                    self.task_sets[*task_sets_index].host.clone(),
```

Comment:
Why do we use & and then dereference here? More curious as I am trying to imagine how it maps to the generated iterator code here.

Reply:
We use `&` so we don't move `weighted_task_sets` when iterating on it.

If we don't use `&` and then dereference, it won't compile because of:

```
error: use of moved value: `weighted_task_sets`
label: value moved here, in previous iteration of loop
```